apps/api/v2/src/ee/bookings/2024-08-13/services/bookings.service.ts

```diff
@@ -990,4 +999,79 @@ export class BookingsService_2024_08_13 {
       t: await getTranslation("en", "common"),
     });
   }
+
+  async updateBooking(bookingUid: string, body: UpdateBookingInput_2024_08_13) {
+    const existingBooking = await this.bookingsRepository.getByUid(bookingUid);
+    if (!existingBooking) {
+      throw new NotFoundException(`Booking with uid=${bookingUid} not found`);
+    }
+    const { location } = body;
+
+    if (!location) {
+      return this.getBooking(existingBooking.uid);
+    }
+
+    return await this.updateBookingLocation(existingBooking, location);
+  }
+
+  async updateBookingLocation(existingBooking: Booking, location: BookingInputLocation_2024_08_13) {
+    const bookingUid = existingBooking.uid;
+    let bookingLocation = "";
+
+    if (existingBooking.userId === null) {
+      throw new NotFoundException(`No user found for booking with uid=${bookingUid}`);
+    }
+
+    if (existingBooking.eventTypeId === null) {
+      throw new NotFoundException(`No event type found for booking with uid=${bookingUid}`);
+    }
+
+    const existingBookingHost = await this.usersRepository.findById(existingBooking.userId);
+
+    if (existingBookingHost === null) {
+      throw new NotFoundException(`No user found for booking with uid=${bookingUid}`);
+    }
+
+    const existingBookingEventType = await this.eventTypesRepository.getEventTypeById(
+      existingBooking.eventTypeId
+    );
+
+    if (typeof location === "object" && location.type === "organizersDefaultApp") {
+      const existingBookingHostCredentials = await this.credentialsRepository.getAllUserCredentialsById(
+        existingBookingHost?.id
+      );
+      const existingBookingUserOrgId = await getOrgIdFromMemberOrTeamId({
+        memberId: existingBooking.userId ?? null,
+        teamId: existingBookingEventType?.teamId,
+      });
+      const enrichedUser = await enrichUsersWithDelegationCredentials({
+        orgId: existingBookingUserOrgId ?? null,
+        users: [
+          {
+            id: existingBookingHost.id,
+            email: existingBookingHost.email,
+            credentials: existingBookingHostCredentials,
+          },
+        ],
+      });
+
+      bookingLocation = getDefaultConferencingAppLocation(
+        existingBookingHost?.metadata,
+        enrichedUser[0].credentials
+      );
+    } else if (typeof location === "object") {
+      const transformedLocation = this.inputService.transformLocation(location);
+      const { bookingLocation: bookingLocationForDB } = getLocationValueForDB(
+        transformedLocation?.optionValue || transformedLocation?.value,
```

Comment:
`transformedLocation?.optionValue || transformedLocation?.value` is taken from `packages/features/bookings/lib/handleNewBooking/getBookingData.ts` line 74:

```
responses.location?.optionValue || responses.location?.value || "",
```

I think it is a good idea to create a new function `getBookingDataLocation` to which we pass responses location or transformedLocation in case of updating and then re-use this function in the `getBookingData` on line 74 and here in this code.

This would reduce the possibility that if creation logic changes that updating logic is out of date and would ensure that the logic is the same.